ui/src/page/start_game.rs

```diff
@@ -64,23 +74,37 @@ pub fn init(session: Session) -> Model {
 pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
     match msg {
         Msg::StartClicked => {
-            let game_request = GameRequest::init(model.game_name_field.clone());
+            let game_request = start_game::Request::init(
+                model.game_name_field.clone(),
+                model.player_name_field.clone(),
+            );
 
             let url = model.get_session_mut().url("/game/create");
 
-            let byte_data = game_request.get_bytes_array();
-            orders.skip().perform_cmd({
-                async {
-                    let response = match send_message(url, byte_data).await {
-                        Ok(_) => Response::Yep,
-                        Err(err) => {
-                            log!(err);
-                            Response::Nope
+            match game_request.to_bytes() {
+                Ok(bytes) => {
+                    model.waiting();
+                    orders.skip().perform_cmd({
+                        async {
+                            let msg = match send_message(url, bytes).await {
+                                Ok(bytes) => match start_game::Response::from_bytes(bytes) {
+                                    Ok(response) => Msg::NewGameResponse(response),
+                                    Err(error) => Msg::NewGameFailed(error.to_string()),
+                                },
+                                Err(error) => {
+                                    let fetch_error = util::http::fetch_error_to_string(error);
+                                    Msg::NewGameFailed(fetch_error)
+                                }
+                            };
+
+                            msg
```

Comment:
These deeply nested matches kind of stink. But I dont know how to avoid that in Rust.